beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/path/aws/S3PathCleaner.java

```diff
@@ -65,14 +65,15 @@ public void cleanupPath(String housekeepingPath, String tableName) {
       deleteFilesInDirectory(bucket, key);
 
       try {
-        if (housekeepingPath.endsWith("/")) {
-          housekeepingPath = housekeepingPath.substring(0, housekeepingPath.length() - 1);
+        String path = s3SchemeURI.getPath();
+        if (s3SchemeURI.getPath().endsWith("/")) {
+          path = path.substring(0, housekeepingPath.length() - 1);
```

Comment:
is this correct? `housekeepingPath.length() - 1`

Reply:
Good spot I missed one :(